src/js/base/module/Editor.js

```diff
@@ -452,14 +452,26 @@ define([
       if (onApplyCustomStyle) {
         onApplyCustomStyle.call(this, $target, context, this.onFormatBlock);
       } else {
-        this.onFormatBlock(tagName);
+        this.onFormatBlock(tagName, $target);
       }
     });
 
-    this.onFormatBlock = function (tagName) {
+    this.onFormatBlock = function (tagName, $target) {
       // [workaround] for MSIE, IE need `<`
       tagName = agent.isMSIE ? '<' + tagName + '>' : tagName;
       document.execCommand('FormatBlock', false, tagName);
+
+      // support custom class 
+      if ($target && $target.length) {
+        var className = $target[0].className || '';
+        if (className) {
+          var currentRange = this.createRange();
+  
+          var $parent = $([currentRange.sc, currentRange.ec]).closest(tagName);
+          $parent.addClass(className);
+        }
+      }
```

Comment:
This logic seems to work only with the bodycontainer type(ex: blockquote). I'm not sure, But If we select multiple blocks(ex: paragraph), className will be applied only first block element.

Could you added a test for multiple paragraphs?

Reply:
i add test unit for multi formatblock.   please check it.